### PR TITLE
Add IPFS buttons to protocol badge popup

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/16907

- Added IPFS buttons to IPFS protocol badge popup

image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Go to any ipfs page
- Click by protocol badge
- Make sure buttons are visible only for ipfs pages

### CODE REVIEW

**Comment by spylogsster** [comment]
@bbondy @karenkliu any proposition about icons maybe?

**Comment by autonome** [comment]
Nice to have request:

Add visual distinction for things that are clickable vs not. Eg, underline links, or make them buttons. Or sometimes I see the icon at the end of the line which is like an arrow, signifying that you click that in order to leave native UI to open a link.

**Comment by bbondy** [comment]
I defer to @karenkliu but if not you can always start with just links

File: components/resources/ipfs_strings.grdp [hunk 1]
```diff
@@ -120,6 +120,24 @@
...
+      Go to IPFS Settings page
+    </message>
+    <message name="IDS_PAGE_INFO_IPFS_DIAGNOSTICS_BUTTON_TEXT" desc="The text for ips page info bubble ipfs diagnostic button text.">
+      IPFS Diagnostics
+    </message>
+    <message name="IDS_PAGE_INFO_IPFS_DIAGNOSTICS_BUTTON_TOOLTIP_TEXT" desc="The text for ips page info bubble diagnostic button tooltip text.">
+      Go to IPFS Diagnostics page
+    </message>
+    <message name="IDS_PAGE_INFO_IPFS_DOCS_BUTTON_TEXT" desc="The text for ips page info bubble ipfs diagnostic button text.">
+      IPFS Docs
```
Comment (bbondy) [inline components/resources/ipfs_strings.grdp:136]: IPFS docs

File: components/resources/ipfs_strings.grdp [hunk 2]
```diff
@@ -120,6 +120,24 @@
     <message name="IDS_PAGE_INFO_IPFS_BUBBLE_TEXT" desc="The text for ips page info bubble.">
       InterPlanetary File System (IPFS) is a public network for peer-to-peer data storage and delivery. You can learn more about Brave's integration or configure your IPFS settings at any time.
     </message>
+    <message name="IDS_PAGE_INFO_IPFS_SETTINGS_BUTTON_TEXT" desc="The text for ips page info bubble IPFS Settings button.">
+      IPFS Settings
```
Comment (bbondy) [inline components/resources/ipfs_strings.grdp:124]: IPFS settings

File: components/resources/ipfs_strings.grdp [hunk 3]
```diff
@@ -120,6 +120,24 @@
...
+    </message>
+    <message name="IDS_PAGE_INFO_IPFS_SETTINGS_BUTTON_TOOLTIP_TEXT" desc="The text for ips page info bubble IPFS Settings button tooltip text.">
+      Go to IPFS Settings page
```
Comment (bbondy) [inline components/resources/ipfs_strings.grdp:127]: Go to the IPFS settings page

File: components/resources/ipfs_strings.grdp [hunk 4]
```diff
@@ -120,6 +120,24 @@
...
+    </message>
+    <message name="IDS_PAGE_INFO_IPFS_DIAGNOSTICS_BUTTON_TEXT" desc="The text for ips page info bubble ipfs diagnostic button text.">
+      IPFS Diagnostics
```
Comment (bbondy) [inline components/resources/ipfs_strings.grdp:130]: IPFS diagnostics

File: components/resources/ipfs_strings.grdp [hunk 5]
```diff
@@ -120,6 +120,24 @@
...
+    </message>
+    <message name="IDS_PAGE_INFO_IPFS_DIAGNOSTICS_BUTTON_TOOLTIP_TEXT" desc="The text for ips page info bubble diagnostic button tooltip text.">
+      Go to IPFS Diagnostics page
```
Comment (bbondy) [inline components/resources/ipfs_strings.grdp:133]: Go to the IPFS diagnostics page

File: components/resources/ipfs_strings.grdp [hunk 6]
```diff
@@ -120,6 +120,24 @@
...
+    </message>
+    <message name="IDS_PAGE_INFO_IPFS_DOCS_BUTTON_TEXT" desc="The text for ips page info bubble ipfs diagnostic button text.">
+      IPFS Docs
+    </message>
+    <message name="IDS_PAGE_INFO_IPFS_DOCS_BUTTON_TEXT_TOOLTIP_TEXT" desc="The text for ips page info bubble docs button tooltip text.">
+      Go to IPFS Docs page
```
Comment (bbondy) [inline components/resources/ipfs_strings.grdp:139]: Go to the IPFS docs page

File: components/resources/ipfs_strings.grdp (hunk 1 above)
Comment (bbondy) [inline components/resources/ipfs_strings.grdp:136]: IPFS docs

**Review by bbondy — APPROVED** [review]
Some minor text changes, but ++

File: chromium_src/chrome/browser/ui/views/page_info/page_info_bubble_view.cc [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+#include "brave/components/vector_icons/vector_icons.h"
+#include "brave/components/ipfs/ipfs_constants.h"
+#include "brave/components/ipfs/ipfs_utils.h"
+#include "brave/grit/brave_theme_resources.h"
+#include "chrome/browser/ui/views/page_info/page_info_bubble_view.h"
+#include "chrome/browser/ui/views/page_info/page_info_view_factory.h"
+#include "components/grit/brave_components_strings.h"
+#include "chrome/browser/ui/browser_tabstrip.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "ui/base/l10n/l10n_util.h"
```
Comment (mkarolin) [inline chromium_src/chrome/browser/ui/views/page_info/page_info_bubble_view.cc:16]: Please, make sure the target already has deps for all these.

**Review by mkarolin — APPROVED** [review]
LGTM, but see the comment on deps.

**Comment by karenkliu** [comment]
LGTM! :)

**Review by lidel — APPROVED** [review]
Good improvement, facilitates self-driven discovery :+1:

**Pull request merged**